rimage/lazy_encoded.go

```diff
@@ -47,6 +51,26 @@ func (lei *LazyEncodedImage) decode() {
 	}
 }
 
+func (lei *LazyEncodedImage) decodeConfig() {
+	lei.decodeConfigOnce.Do(func() {
+		defer func() {
+			if err := recover(); err != nil {
+				lei.decodeErr = err
```

Comment:
actually, I think that if someone tries to call decodeConfig and decode at the same time, there might be a race condition to write over the decodeErr field. So create a new field called lei.decodeConfigErr, and put the error in there